### PR TITLE
Have TypeScript target CommonJS using Browserify

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
The current comparisons turn out to be a little unfair because TypeScript is tested on its AMD output with Uglify. It looks like the bootstrapping and AMD prologues add quite a bit to the output.

This PR removes the AMD loader and bootstrap code, has TypeScript emit as a CommonJS dependency, and runs that through Browserify as an intermediate step.

With these changes, I get

```
-----------------------------------------
stat -c %s src/dist/bundle.js
14980
gzip -c src/dist/bundle.js | wc -c
3997
-----------------------------------------
```

### CODE REVIEW

File: typescript/package.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {
   "scripts": {
-    "compile": "(cat almond.js; ./node_modules/.bin/tsc --allowJs --module amd --outFile /dev/stdout ../src/src/app.js; cat bootstrap.js) | ./node_modules/.bin/uglifyjs --compress - > ../src/dist/bundle.js"
+    "compile": "./node_modules/.bin/tsc && (./node_modules/.bin/browserify ./dist/*.js -o ../src/dist/app.js | ./node_modules/.bin/uglifyjs ../src/dist/app.js --compress -o ../src/dist/bundle.js)"
```
Comment (samccone) [inline typescript/package.json]: I think you will also want to pass --mangle here :) based on the findings in #17

File: typescript/package.json [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
   },
   "dependencies": {
     "typescript": "^1.8.0",
-    "uglifyjs": "^2.4.10"
+    "uglifyjs": "^2.4.10",
```
Comment (samccone) [inline typescript/package.json:7]: mind updating shrinkwrap as well?

Comment (samccone) [inline typescript/package.json:7]: (npm shrinkwrap should do it)

**Comment by samccone** [comment]
just a few comments @DanielRosenwasser thanks a ton for the PR

I will land this commit along with the other ones tonight and update the graphs accordingly

File: typescript/package.json (hunk 1 above)
Comment (samccone) [inline typescript/package.json]: (it should further shrink your bundle)

Comment (DanielRosenwasser) [inline typescript/package.json]: Nice!

```
-----------------------------------------
stat -c %s src/dist/bundle.js
11442
gzip -c src/dist/bundle.js | wc -c
3385
-----------------------------------------
```

Comment (samccone) [inline typescript/package.json]: awwww snap :clap:

**Comment by samccone** [comment]
Hi @DanielRosenwasser 

so it seems like this example is no longer functional after this patch

screen shot 2016-02-02 at 5 59 52 pm

---

(you are going to want to prob want to drop the mangle commit due to my findings here https://github.com/samccone/The-cost-of-transpiling-es2015-in-2016/pull/17#issuecomment-178955003)

**Comment by samccone** [comment]
manually landed in https://github.com/samccone/The-cost-of-transpiling-es2015-in-2016/tree/mangle

going to ship to master with the updated assets soon

thanks again!

**Pull request closed without merging**